src/pages/settings/Profile/Contacts/ContactMethodDetailsPage.tsx

```diff
@@ -265,20 +265,56 @@ function ContactMethodDetailsPage({route}: ContactMethodDetailsPageProps) {
                     />
                 </OfflineWithFeedback>
             )}
-            {getDeleteConfirmationModal()}
         </>
     );
 
     return (
         <ScreenWrapper
-            onEntryTransitionEnd={() => validateCodeFormRef.current?.focus?.()}
+            shouldEnableMaxHeight
+            onEntryTransitionEnd={() => {
+                InteractionManager.runAfterInteractions(() => {
+                    validateCodeFormRef.current?.focus?.();
+                });
+            }}
             testID={ContactMethodDetailsPage.displayName}
+            focusTrapSettings={{
+                focusTrapOptions: isMobileSafari()
+                    ? undefined
+                    : {
+                          // It is added because input form's focusing bothers transition animation:
+                          // https://github.com/Expensify/App/issues/53884#issuecomment-2594568960
+                          checkCanFocusTrap: (trapContainers: Array<HTMLElement | SVGElement>) => {
+                              return new Promise((resolve) => {
+                                  const interval = setInterval(() => {
+                                      const trapContainer = trapContainers.at(0);
+                                      if (!trapContainer || (trapContainer && getComputedStyle(trapContainer).visibility !== 'hidden')) {
```

Comment:
```suggestion
                                      if (!trapContainer || getComputedStyle(trapContainer).visibility !== 'hidden') {
```
`trapContainer` has to exist in the second half of this